chain/evm/deploy_contract.go

```diff
@@ -147,3 +285,70 @@ func deployContract(
 	})
 	return
 }
+
+func (c *Client) wrapArbitrumDeployment(
+	ctx context.Context,
+	chainID *big.Int,
+	rawABI string,
+	bytecode,
+	constructorInput []byte,
+) (contractAddr common.Address, tx *ethtypes.Transaction, err error) {
```

Comment:
This takes the input types from `go-ethereum`, wraps them in the Arbitrum `go-ethereum` types, and then back again after the contract has been deployed. It's messy, but it works.